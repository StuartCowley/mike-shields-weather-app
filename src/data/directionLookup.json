{
  "n": {
    "degrees": 0,
    "humanReadable": "north"
  },
  "nne": {
    "degrees": 22.5,
    "humanReadable": "north north east"
  },
  "ne": {
    "degrees": 45,
    "humanReadable": "north east"
  },
  "ene": {
    "degrees": 67.5,
    "humanReadable": "east north east"
  },
  "e": {
    "degrees": 90,
    "humanReadable": "east"
  },
  "ese": {
    "degrees": 112.5,
    "humanReadable": "north"
  },
  "se": {
    "degrees": 135,
    "humanReadable": "south east"
  },
  "sse": {
    "degrees": 157.5,
    "humanReadable": "se"
  },
  "s": {
    "degrees": 180,
    "humanReadable": "south"
  },
  "ssw": {
    "degrees": 202.5,
    "humanReadable": "south south west"
  },
  "sw": {
    "degrees": 225,
    "humanReadable": "south west"
  },
  "wsw": {
    "degrees": 247.5,
    "humanReadable": "west south west"
  },
  "w": {
    "degrees": 270,
    "humanReadable": "west"
  },
  "wnw": {
    "degrees": 292.5,
    "humanReadable": "west north west"
  },
  "nw": {
    "degrees": 315,
    "humanReadable": "north west"
  },
  "nnw": {
    "degrees": 337.5,
    "humanReadable": "north nort west"
  }
}

