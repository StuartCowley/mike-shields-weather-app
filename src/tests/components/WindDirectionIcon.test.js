import React from "react";
import { render, screen } from "@testing-library/react";
import WindDirectionIcon from "../../components/WindDirectionIcon";
import directionLookup from "../../data/directionLookup.json";

describe("WindDirectionIcon", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<WindDirectionIcon direction="n" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders an icon pointing in the provided direction", async () => {
    Object.keys(directionLookup).forEach(async (direction) => {
      render(<WindDirectionIcon direction={direction} />);

      const WindDirectionIconComponent = await screen.findByTestId(
        "wind-direction-icon"
      );

      expect(WindDirectionIconComponent).toHaveStyle(
        `transform: rotate(${directionLookup[direction].degrees}deg)`
      );
    });
  });

  it("renders an icon with an alt attribute describing the direction of the icon", () => {
    Object.keys(directionLookup).forEach(async (direction) => {
      render(<WindDirectionIcon direction={direction} />);

      const description = new RegExp(directionLookup[direction.humanReadable]);

      const WindDirectionIconComponent = await screen.findByAltText(
        description,
        "i"
      );

      expect(WindDirectionIconComponent).toBeInTheDocument();
    });
  });
});
