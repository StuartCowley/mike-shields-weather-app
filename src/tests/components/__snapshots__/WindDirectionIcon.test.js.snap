// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WindDirectionIcon matches snapshot 1`] = `
<DocumentFragment>
  <img
    alt="arrow pointing undefined"
    class="wind-direction-icon"
    data-testid="wind-direction-icon"
    src="arrow-circle.png"
    style="transform: rotate(0deg);"
  />
</DocumentFragment>
`;
